internal/common/resource.go

```diff
@@ -93,8 +97,45 @@ func createOrUpdate(request *Request, resource controllerutil.Object, isClusterR
 
 func setOwner(request *Request, resource controllerutil.Object, isClusterRes bool) error {
 	if isClusterRes {
-		return libhandler.SetOwnerAnnotations(request.Instance, resource)
+		err := libhandler.SetOwnerAnnotations(request.Instance, resource)
+		if err != nil {
+			return err
+		}
+
+		// Removing ownerReferences for objects prior to creation
+		resource.SetOwnerReferences(nil)
+
+		// Removing ownerReferences for existing objects
+		existingRes := newEmptyResource(resource)
+		key, err := client.ObjectKeyFromObject(resource)
+		if err != nil {
+			return err
+		}
+
+		err = request.Client.Get(request.Context, key, existingRes)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return err
+			}
+		} else {
+			if existingRes.GetOwnerReferences() != nil && len(existingRes.GetOwnerReferences()) > 0 {
+				patch := client.RawPatch(types.JSONPatchType,
+					[]byte(`[{"op": "replace","path": "/metadata/ownerReferences", "value": []}, {"op": "remove","path": "/metadata/ownerReferences"}]`))
```

Comment:
Do we need to replace and remove?
It seems redundant or am I missing something here?

Reply:
Simply removing gave inconsistent errors (both when `ownerReferences` exist and not), so the workaround I chose here is to first `replace` with an empty slice (create if doesn't exist), then `remove`, this seemed to give a consistent error flow that I can follow

Reply:
Makes sense, thanks.